src/puffin/src/puffin_manager/cache.rs

```diff
@@ -27,8 +27,9 @@ pub struct PuffinMetadataCache {
     cache: moka::sync::Cache<String, Arc<FileMetadata>>,
 }
 
+#[allow(clippy::ptr_arg)]
 fn puffin_metadata_weight(k: &String, v: &Arc<FileMetadata>) -> u32 {
-    (k.as_bytes().len() + v.memory_usage()) as u32
+    (k.len() + v.memory_usage()) as u32
 }
```

Comment:
Clippy complains about changing `&String` to `&str`. However, if we do so it then complains about `k` in `.eviction_listener(|k, v, _cause| {` doesn't have a size known at compile-time. So I just add `#[allow(clippy::ptr_arg)]` to pass the check 😢 @zhongzc

Reply:
We can use `&str` for key while changing line 41 to 

```rust
weigher(|k: &String, v| puffin_metadata_weight(k, v))
```